Source/WebCore/css/StyleColor.cpp

```diff
@@ -77,8 +78,21 @@ bool StyleColor::isSystemColorKeyword(CSSValueID id)
 bool StyleColor::isColorKeyword(CSSValueID id, OptionSet<CSSColorType> allowedColorTypes)
 {
     return (allowedColorTypes.contains(CSSColorType::Absolute) && isAbsoluteColorKeyword(id))
-        || (allowedColorTypes.contains(CSSColorType::Current) && id == CSSValueCurrentcolor)
+        || (allowedColorTypes.contains(CSSColorType::Current) && isCurrentColorKeyword(id))
         || (allowedColorTypes.contains(CSSColorType::System) && isSystemColorKeyword(id));
 }
 
+bool StyleColor::isCurrentColorKeyword(CSSValueID id)
+{
+    return (id == CSSValueCurrentcolor);
+}
+
+bool StyleColor::isCurrentColor(const CSSPrimitiveValue& value)
+{
+    if (value.isValueID())
```

Comment:
No need to check isValueID() here. valueID() returns CSSValueInvalid for the case when it is not.